### ISSUE TITLE
3.02 Videosource needed

### ISSUE THREAD

**Issue by jmcexx** [GerryDazoo/Slinger#14]
Hi Gerry, 

  First, letting you know your 3.02 is working great on PRO-HD.  Sorry the delay testing.

  Want to let you know I kept getting the following error message and the server would reset each time I tried to connect with 3.02.  I added a VideoStream=0 line to the config.ini file and I no longer get the error, and it works.   Didn't have any problem with 3.01.

- Ryan 

```
Aug 06 08:01:59 ITAM-473 python3[12637]: Traceback (most recent call last):
Aug 06 08:01:59 ITAM-473 python3[12637]:   File "slingbox_server.py", line 548, in streamer
Aug 06 08:01:59 ITAM-473 python3[12637]:     s_ctl, stream  = start_slingbox_session(streams)
Aug 06 08:01:59 ITAM-473 python3[12637]:   File "slingbox_server.py", line 325, in start_slingbox_session
Aug 06 08:01:59 ITAM-473 python3[12637]:     while dbuf[i] != 0 :
Aug 06 08:01:59 ITAM-473 python3[12637]: IndexError: index out of range
Aug 06 08:01:59 ITAM-473 python3[12637]:  Badness starting slingbox session  index out of range None
```

**Comment by GerryDazoo** [comment]
Thanks for the report. I’ll look into it. It’s new code that dumps out the remote control key codes configured for the video source. Can you grab the logs that were printed just before the error.

 

*	Gerry

 

From: jmcexx ***@***.***> 
Sent: Saturday, August 06, 2022 9:15 AM
To: GerryDazoo/Slinger ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GerryDazoo/Slinger] 3.02 Videosource needed (Issue #14)

 

Hi Gerry,

First, letting you know your 3.02 is working great on PRO-HD. Sorry the delay testing.

Want to let you know I kept getting the following error message and the server would reset each time I tried to connect with 3.02. I added a VideoStream=0 line to the config.ini file and I no longer get the error, and it works. Didn't have any problem with 3.01.

*	Ryan

Aug 06 08:01:59 ITAM-473 python3[12637]: Traceback (most recent call last):
Aug 06 08:01:59 ITAM-473 python3[12637]:   File "slingbox_server.py", line 548, in streamer
Aug 06 08:01:59 ITAM-473 python3[12637]:     s_ctl, stream  = start_slingbox_session(streams)
Aug 06 08:01:59 ITAM-473 python3[12637]:   File "slingbox_server.py", line 325, in start_slingbox_session
Aug 06 08:01:59 ITAM-473 python3[12637]:     while dbuf[i] != 0 :
Aug 06 08:01:59 ITAM-473 python3[12637]: IndexError: index out of range
Aug 06 08:01:59 ITAM-473 python3[12637]:  Badness starting slingbox session  index out of range None

**Comment by GerryDazoo** [comment]
It just uploaded V3.03 that should fix this issue. Please give it a try

 

From: jmcexx ***@***.***> 
Sent: Saturday, August 06, 2022 9:15 AM
To: GerryDazoo/Slinger ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GerryDazoo/Slinger] 3.02 Videosource needed (Issue #14)

 

Hi Gerry,

First, letting you know your 3.02 is working great on PRO-HD. Sorry the delay testing.

Want to let you know I kept getting the following error message and the server would reset each time I tried to connect with 3.02. I added a VideoStream=0 line to the config.ini file and I no longer get the error, and it works. Didn't have any problem with 3.01.

*	Ryan

Aug 06 08:01:59 ITAM-473 python3[12637]: Traceback (most recent call last):
Aug 06 08:01:59 ITAM-473 python3[12637]:   File "slingbox_server.py", line 548, in streamer
Aug 06 08:01:59 ITAM-473 python3[12637]:     s_ctl, stream  = start_slingbox_session(streams)
Aug 06 08:01:59 ITAM-473 python3[12637]:   File "slingbox_server.py", line 325, in start_slingbox_session
Aug 06 08:01:59 ITAM-473 python3[12637]:     while dbuf[i] != 0 :
Aug 06 08:01:59 ITAM-473 python3[12637]: IndexError: index out of range
Aug 06 08:01:59 ITAM-473 python3[12637]:  Badness starting slingbox session  index out of range None

**Comment by jmcexx** [comment]
Yep, 3.03 works without the videosource line now.  However, it doesn't print out the key codes without the videosource line.  Which is a really great addition you made.  Not a problem, just letting you know.  thanks.

**Comment by jmcexx** [comment]
Would you be able to check all 4 input sources?  You'd know which inputs have remotes associated with them maybe.

**Comment by GerryDazoo** [comment]
That would be a major complication. It should work now with the VideoSource set.

**Comment by GerryDazoo** [comment]
FYI, the slingbox requires a videosource so it can return the proper remote info for that configured source. That's why I require you to set the video source in the config.ini to get the key codes. I don't know in advance which and how many sources you may have configured. You can always remove the config.ini entry after you've recovered the keycodes you're looking for. But I don't see a problem in leaving the VideoSource permanently set in the config file.

**Comment by jmcexx** [comment]
I was only thinking where you print the key codes for the one video source, it could print for all 4.  But I can see how that could be confusing also.  Anyways, works great with videosource set.  The current config.ini file on repo doesn't have a videosource line, though.

**Comment by GerryDazoo** [comment]
I'll make that change.

**Comment by GerryDazoo** [comment]
Fixed in V3.03
